src/core/Akka.Tests/Actor/AskTimeoutSpec.cs

```diff
@@ -56,16 +56,24 @@ public AskTimeoutSpec()
             var actor = Sys.ActorOf<SleepyActor>();
 
             var actorCell = actor as ActorRefWithCell;
+            Assert.NotNull(actorCell);
+
             var container = actorCell.Provider.TempContainer as VirtualPathContainer;
+            Assert.NotNull(container);
             try
             {
                 await actor.Ask<string>("should time out");
             }
             catch (Exception)
             {
-                var childCounter = 0;
-                container.ForEachChild(x => childCounter++);
-                Assert.True(childCounter==0,"Number of children in temp container should be 0.");
+                // Need to spin here, since the continuation function may not execute immediately
+                AwaitAssert(() =>
```

Comment:
Actual fix: the `FutureActorRef` is not unregistered from the `VirtualPathContainer` until after a continuation function has a chance to run, hence we need an `AwaitAssert` here in case that continuation hasn't been scheduled by the time we perform the assertion the first time around.